app/workers/enqueue_work_package_notification_job.rb

```diff
@@ -76,6 +78,46 @@ def work_package
   end
 
   def notification_receivers(work_package)
-    (work_package.recipients + work_package.watcher_recipients).uniq
+    (work_package.recipients + work_package.watcher_recipients + mentioned).uniq
   end
+
+  def mentioned
+    mentioned_people = []
+
+    text = text_for_mentions
+    user_ids = text.scan(/\buser#([\d]+)\b/).first.to_a
+    user_login_names = text.scan(/\buser:"(.+?)"/).first.to_a
+
+    user_ids.each do |user_id|
+      if user = User.find_by(id: user_id)
+        if @work_package.visible? user
+          mentioned_people << user unless user.mail_notification == 'none'
+        end
+      end
+    end
+
+    user_login_names.each do |name|
+      if user = User.find_by(login: name)
+        if @work_package.visible? user
+          mentioned_people << user unless user.mail_notification == 'none'
+        end
+      end
+    end
+    mentioned_people
+  end
+
+  def text_for_mentions
+    potential_text = ""
+    potential_text << @journal.notes if @journal.try(:notes)
+
+    [:description, :subject].each do |field|
+      if @journal.details[field].try(:any?)
```

Comment:
If I understand the flow here, `@journal` may still be nil at this point, right?

Reply:
No, ```@journal``` gets initialized in in ```perform```. If it turns out to be nil the code does not continue. See here:
https://github.com/opf/openproject/blob/dev/app/workers/enqueue_work_package_notification_job.rb#L47

Reply:
Dang, you're right 💃